internal/controller/thanosruler_controller.go

```diff
@@ -326,6 +329,44 @@ func (r *ThanosRulerReconciler) getPrometheusRuleConfigMaps(ctx context.Context,
 		"ruler", ruler.Name,
 		"namespace", ruler.Namespace)
 
+	if ruler.Spec.RuleTenancyConfig != nil {
+		tenantLabel := ruler.Spec.RuleTenancyConfig.TenantLabel
+		tenantValueLabel := ruler.Spec.RuleTenancyConfig.TenantValueLabel
+
+		// Modify PrometheusRule objects to include tenant label
+		for _, rule := range promRules.Items {
+			if rule.Labels == nil {
+				rule.Labels = make(map[string]string)
+			}
+			value, exists := rule.Labels[tenantValueLabel]
```

Comment:
@saswatamcode what if `value` is empty? Shouldn't it be treated as an error/misconfiguration?

Reply:
So here we don't really consider it as misconfig, but we do log it.

This pass through provides us with the ability to have certain cross-tenant rules, which aren't locked to any particular tenant.

Reply:
Oh! That's a great catch, let me fix!